From 5ca658e2a616275e592704e02dd8525d29059812 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Mon, 5 Aug 2013 10:18:24 +0000
Subject: [PATCH] Instead of relying on existing service/role directories
 create them on startup

It makes the service start a lot easier when the hub daemon takes care about creating the
service/role directories on its own instead of doing this job as part of the init script
(which is very close to the daemon anyway).

Upstream-Status: pending (http://github.com/openwebos/luna-service2)

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/ls-hubd/hub.c      | 6 ++++++
 src/ls-hubd/security.c | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/src/ls-hubd/hub.c b/src/ls-hubd/hub.c
index 23d8875..b70b628 100644
--- a/src/ls-hubd/hub.c
+++ b/src/ls-hubd/hub.c
@@ -1151,6 +1151,12 @@ ParseServiceDirectory(const char *path, LSError *lserror)
 
     _ls_verbose("%s: parsing service directory: \"%s\"\n", __func__, path);
 
+    if (g_file_test(path, G_FILE_TEST_EXISTS) == FALSE)
+    {
+        g_warning("WARNUNG: service directory \"%s\" does not exist. Attempting to create it ...", path);
+        g_mkdir_with_parents(path, 0777);
+    }
+
     GDir *dir = g_dir_open(path, 0, &gerror);
 
     if (!dir)
diff --git a/src/ls-hubd/security.c b/src/ls-hubd/security.c
index b70f43f..9d0816d 100644
--- a/src/ls-hubd/security.c
+++ b/src/ls-hubd/security.c
@@ -1239,6 +1239,12 @@ ParseRoleDirectory(const char *path, GHashTable *role_hash, GHashTable *perm_has
 
     _ls_verbose("%s: parsing role directory: \"%s\"\n", __func__, path);
 
+    if (!g_file_test(path, G_FILE_TEST_EXISTS))
+    {
+        g_warning("WARNING: role directory \"%s\" does not exist. Attempting to create it ...", path);
+        g_mkdir_with_parents(path, 0777);
+    }
+
     GDir *dir = g_dir_open(path, 0, &gerror);
 
     if (!dir)
-- 
1.8.1.2

